test/README.md

```diff
@@ -0,0 +1,158 @@
+Running DMD's test suite
+-------------------------
+
+This is the documentation and guide for DMD's test suite.
+Maybe you are looking for the [contributing guide](../CONTRIBUTING.md) too?
+
+Quick guide
+-----------
+
+### Run all tests
+
+```sh
+make -j10
+```
+
+Note: the exact parallelism parameter `-j` depends on your system,
+but if you have enough memory, a high `-j` allows a significant speed-up.
+
+### Run only a specific subset
+
+```sh
+make -j10 run_fail_compilation_tests
+make -j10 run_compilable_tests
+```
+
+As linking is slow the `runnable` tests take a bit longer:
+
+```sh
+make -j10 run_runnable_tests
+```
+
+### Run only an individual test
+
+```sh
+./run_individual_tests fail_compilation/diag10089.d
+```
+
+Multiple arguments are supported too.
+You can use `./run_individual_tests` to quickly run a custom subset of tests.
+For example, all diagnostic tests in `fail_compilation`:
+
+```sh
+./run_individual_tests fail_compilation/diag*.d
+```
+
+### Automatically update the error messages
```

Comment:
I don't understand what this means.  What error messages is it updating and why would someone want to do that?

Reply:
Ah, you mean this automatically overwrites the TEST_OUTPUT for the given tests?